docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
+1. If you're starting from scratch, you can use the [`brew create` command](Manpage.md#create-options-url) to produce a basic version of your formula. This command accepts a number of options and you may be able to save yourself some work by using an appropriate template option like `--python` / `--go`.
 
-1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Cookbook for help with this.
+1. You will now have to develop the boilerplate code from `brew create` into a complete formula. Your main references will be the [Formula Cookbook](Formula-Cookbook.md), similar existing formulae, and the official documentation of the software you're packaging. Be sure to also take note of the Homebrew documentation for writing [Python](Python-for-Formula-Authors.md) and [Node](Node-for-Formula-Authors.md) formulae, if applicable.
 
-1. Try installing your formula using `brew install --build-from-source <formula>`, where *\<formula>* is the name of your formula. If any errors occur, correct your formula and attempt to install it again. The formula installation should finish without errors by the end of this step.
+1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Formula Cookbook for help with this.
 
 If you're stuck, ask for help on GitHub or the [Homebrew discussion forum](https://github.com/orgs/Homebrew/discussions). The maintainers are very happy to help but we also like to see that you've put effort into trying to find a solution first.
 
 ### Testing and auditing the formula
 
-1. Run `brew audit --strict --new-formula --online <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+1. Test your formula installation using `brew install --formula --build-from-source $(brew --repository homebrew/core)/Formula/<prefix>/<formula>.rb`, where `<prefix>` is the first letter of your formula's name and `<formula>` is the name of your formula. Correct any errors that occur and repeat the installation until it completes without errors.
+
+1. Additionally, verify the proper uninstallation of the formula by using the command `brew uninstall --formula <formula>`, ensuring it removes the software without issues.
+
+1. Run `brew audit --formula --new <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+
+1. Run `brew style --formula --fix <formula>` to automatically check and correct your formula's conformity to Homebrew's style guidelines.
```

Comment:
`brew style` is already run by `brew audit` so I'd suggest this be put before `brew audit`.

Reply:
Probably also worth explicitly mentioning that as an offhand comment here too; so people know that if they're about to run the `audit`, they don't need to run the `style` first.

---

Semi-related:

> Since it seems `brew audit` will also run style checks, it might be useful to also mention that somewhere explicitly in it's help text; so that users know they don't need to run `brew style --fix foo` immediately followed by `brew audit --new foo`; when they could have just run `brew audit --fix --new foo` or similar (maybe this is something that can live in the larger documentation rather than the commands `--help` text; eg. https://github.com/Homebrew/brew/pull/16327 or similar; see specifically: https://github.com/Homebrew/brew/pull/16327#discussion_r1425267157)
> 
> https://github.com/Homebrew/brew/blob/2e9d7fbcf81134fbe594d5a18b5387dad76383a1/Library/Homebrew/dev-cmd/audit.rb#L199-L200
> 
>_Originally posted by @0xdevalias in https://github.com/Homebrew/brew/issues/16334_

Reply:
@MikeMcQuaid Fixed in a6d4dba3ab6d69ab3966c6eb64bdb7e663d56244

I'll leave this comment thread open for visibility for now; but once you check it and are happy, leave a comment and then feel free to mark it resolved.